packages/frontend/components/Shared/ConfirmTransactionModal.tsx

```diff
@@ -0,0 +1,301 @@
+  const { editedPosition, position } = basePosition;
+
+  const dynamicLtvMeta = {
+    ltv: editedPosition ? editedPosition.ltv : position.ltv,
+    ltvMax: position.ltvMax,
+    ltvThreshold: editedPosition
+      ? editedPosition.ltvThreshold
+      : position.ltvThreshold,
+  };
+
+  const estCost =
+    transactionMeta.status === 'ready'
+      ? `~$${transactionMeta.bridgeFee.toFixed(2)} + gas`
+      : 'n/a';
+
+  const positionBorrowLimit = remainingBorrowLimit(
+    position.collateral,
+    position.debt,
+    position.ltvMax
+  );
+
+  const editedBorrowLimit =
+    editedPosition &&
+    remainingBorrowLimit(
+      editedPosition.collateral,
+      editedPosition.debt,
+      position.ltvMax
+    );
+
+  const getLtv = (value: number): string => {
+    return value <= 100 && value >= 0 ? `${value.toFixed(0)}%` : 'n/a';
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onClose}
+      sx={{
+        '& .MuiPaper-root': {
+          '-ms-overflow-style': 'none',
+          'scrollbar-width': 'none',
+          '&::-webkit-scrollbar': {
+            display: 'none',
+          },
+        },
+        '& .MuiDialog-paper': {
+          maxWidth: '35rem',
+          width: { xs: '80%', sm: '35rem' },
+        },
+      }}
+    >
+      <Paper
+        variant="outlined"
+        sx={{
+          width: { xs: 'auto', sm: '35rem' },
+          p: { xs: '1rem', sm: '1.5rem' },
+          textAlign: 'center',
+        }}
+      >
+        <CloseIcon
+          sx={{
+            cursor: 'pointer',
+            position: 'absolute',
+            right: '3%',
+          }}
+          onClick={onClose}
+        />
+        <Typography mb="1rem" variant="h5" color={palette.text.primary}>
+          Confirm Transaction
+        </Typography>
+
+        <Stack>
+          {collateral.input && collateral.input !== '0' ? (
+            <AssetBox
+              type="collateral"
+              isEditing={isEditing}
+              actionType={actionType}
+              token={collateral.token}
+              value={collateral.input}
+            />
+          ) : (
+            <></>
+          )}
+
+          {debt.input && debt.input !== '0' ? (
+            <AssetBox
+              type="debt"
+              isEditing={isEditing}
+              actionType={actionType}
+              token={debt.token}
+              value={debt.input}
+            />
+          ) : (
+            <></>
+          )}
+        </Stack>
+
+        {steps && steps.length > 0 && <RouteBox steps={steps} />}
+
+        <InfoRow
+          title="Estimated Cost"
+          value={<Typography variant="small">{estCost}</Typography>}
+        />
+
+        <InfoRow
+          title="Est.processing time"
+          value={
+            <Typography variant="small">{`~${
+              transactionMeta.estimateTime / 60
+            } minutes`}</Typography>
+          }
+        />
+
+        {collateral.chainId !== debt.chainId && (
```

Comment:
This check is also poor @NikolaiYurchenko , need to change it according to Boyan's comment.